subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultArtifactTransforms.java

```diff
@@ -35,8 +36,8 @@ public DefaultArtifactTransforms(
         this.attributesFactory = attributesFactory;
     }
 
-    public VariantSelector variantSelector(AttributeContainerInternal consumerAttributes, boolean allowNoMatchingVariants) {
-        return new AttributeMatchingVariantSelector(consumerProvidedVariantFinder, schema, attributesFactory, consumerAttributes.asImmutable(), allowNoMatchingVariants);
+    public VariantSelector variantSelector(AttributeContainerInternal consumerAttributes, boolean allowNoMatchingVariants, ConfigurationInternal configuration) {
+        return new AttributeMatchingVariantSelector(consumerProvidedVariantFinder, schema, attributesFactory, consumerAttributes.asImmutable(), allowNoMatchingVariants, configuration);
```

Comment:
I see that we are now pushing the `Configuration` object down to the transform across half the world. It would really clarify things if we could a) call the parameter/variable something that designates its role and purpose instead of repeating the type name, or even better b) pass down only a type that exposes only the necessary functionality, and give that type a good name.

Reply:
Half the world ... it is a very short path compared to others 😉 
But you are perfectly right about both the type and the name ... as soon as we know exactly what will be required to get the `Set<ResolvedArtifactResult>`